graphql_compiler/schema_transformation/utils.py

```diff
@@ -254,18 +254,18 @@ def check_schema_identifier_is_valid(identifier: str) -> None:
         )
 
 
-def type_name_is_valid(name: str) -> bool:
-    """Check if input is a valid, nonreserved GraphQL type name.
+def is_valid_unreserved_name(name: str) -> bool:
+    """Check if input is a valid, unreserved GraphQL name.
```

Comment:
minor preference: `nonreserved` (maybe with a hyphen?) rather than `unreserved` since unreserved already has a different meaning in english.